packages/codemods/src/plugins/types.ts

```diff
@@ -40,6 +44,20 @@ export type BuildRefAttributesContext<T> = ValidateConfig<
   }
 >;
 
+export type BuildAttributesContext<T> = ValidateConfig<
+  T,
+  {
+    // The original component declaration
+    originalNode: ASTPath<VariableDeclaration | TaggedTemplateExpression>;
+    // The existing component declaration
+    currentNode: ASTPath<VariableDeclaration | TaggedTemplateExpression>;
+    // The original node after transforms
+    transformedNode: VariableDeclaration | TaggedTemplateExpression | ArrowFunctionExpression;
+    // The composed node that's been created during transformation
+    composedNode: VariableDeclaration | TaggedTemplateExpression | null;
+  }
```

Comment:
maybe it's me but these comments don't help :) Not sure what the difference is between `original` and `exiting` nodes. Also each property has `Node` suffix but 2 of them are `ASTPath`s :) Again, maybe it's just me

Reply:
so, `original` node is the node without any transformations and the `current` node is the node after applying plugins consecutively. You also have an access to a transformed node after initial transformations but without any plugins applied to it yet.